src/components/picker/types.tsx

```diff
@@ -147,49 +122,111 @@ export type PickerBaseProps = Omit<NewTextFieldProps, 'value' | 'onChange'> & {
    * Render custom search input (only when passing showSearch)
    */
   renderCustomSearch?: (props: PickerItemsListProps) => React.ReactElement;
+};
+
+type PickerExpandableOverlayProps = {
   /**
-   * Render a custom header for Picker's dialog
+   * Custom picker overlay props
    */
-  renderCustomDialogHeader?: (callbacks: {onDone?: () => void, onCancel?: ()=> void}) => React.ReactElement;
+  customPickerProps?: ExpandableOverlayProps;
   /**
-   * Use wheel picker instead of a list picker
+   * Render custom picker overlay (e.g ({visible, children, toggleModal}) => {...})
    */
-  useWheelPicker?: boolean;
+  renderOverlay?: (modalProps: RenderCustomModalProps) => React.ReactElement;
   /**
-   * Pass props to the list component that wraps the picker options (allows to control FlatList behavior)
+   * Add blur effect to picker modal (iOS only)
    */
-  listProps?: Partial<FlatListProps<any>>;
+  enableModalBlur?: boolean;
+};
+
+type PickerListProps = PickerSearchProps & {
```

Comment:
What about 'PickerItemsListProps'? Is not a part of this type?

Reply:
As we spoke I'll rearrange it but we will leave it this way.